internal/server/config/config.go

```diff
@@ -31,6 +31,10 @@ const (
 	// care should be taken here - decreasing can cause API compatibility problems with Conditions
 	DefaultMaxConditionEvaluationCost = 100
 	DefaultInterruptCheckFrequency    = 100
+
+	DefaultDispatchThrottlingEnabled             = false
+	DefaultDispatchThrottlingTimeTickerFrequency = 10 * time.Microsecond
```

Comment:
This is likely too aggressive. Most OS' won't be able to provide this level of precision on the timer. 1ms is about the highest precision resolution in general on Unix systems, but on windows it's much worse.

See https://pkg.go.dev/time#hdr-Timer_Resolution

We may consider moving this number up to 1.5-5ms range?

Reply:
Based on my POC test, if this is in the ms range, the throttling becomes very aggressive and "bad" requests will timeout very quickly.  Imagine in the case where 
- check requests timeout in 3 seconds (i.e., 3000ms) 
- you have 7 check request with 500 dispatches 
- throttling start at 50 dispatches

450 * 7 =   3150 (or 3.15s) 
which means some of these requests will timeout even without counting other delays.